SimPEG/ObjectiveFunction.py

```diff
@@ -0,0 +1,258 @@
+            )
+
+        if isinstance(self, ComboObjectiveFunction):
+
+            if isinstance(objfct2, ComboObjectiveFunction):
+                objfctlist = self.objfcts + objfct2.objfcts
+                multipliers = self.multipliers + objfct2.multipliers
+
+            elif isinstance(objfct2, ObjectiveFunction):
+                objfctlist = self.objfcts.append(objfct2)
+                multipliers = self.multipliers.append(1)
+
+        else:
+            if isinstance(objfct2, ComboObjectiveFunction):
+                objfctlist = [self] + objfct2.objfcts
+                multipliers = [1] + objfct2.multipliers
+
+            else:
+                objfctlist = [self, objfct2]
+                multipliers = None
+
+        return ComboObjectiveFunction(
+            objfcts=objfctlist, multipliers=multipliers
+        )
+
+    def __radd__(self, objfct2):
+        return self+objfct2
+
+    def __mul__(self, multiplier):
+        return ComboObjectiveFunction([self], [multiplier])
+
+    def __rmul__(self, multiplier):
+        return self * multiplier
+
+    def __div__(self, denominator):
+        return self.__mul__(1./denominator)
+
+
+class ComboObjectiveFunction(BaseObjectiveFunction):
+
+    _multiplier_types = (float, None, Utils.Zero) + integer_types # Directive
+
+    def __init__(self, objfcts=[], multipliers=None, **kwargs):
+
+        self.objfcts = []
+        nP = '*'
+
+        for fct in objfcts:
+            assert isinstance(fct, BaseObjectiveFunction), (
+                "Unrecognized objective function type {} in objfcts. All "
+                "entries in objfcts must inherit from "
+                "ObjectiveFunction".format(fct.__class__.__name__)
+            )
+
+            # ensure all objective functions have the same nP
+            if fct.nP != '*':
+                if nP != '*':
+                    assert nP == fct.nP, (
+                        "Objective Functions must all have the same nP, not "
+                        "{}".format([f.nP for f in objfcts])
+                    )
+                else:
+                    nP = fct.nP
+            self.objfcts.append(fct)
+
+        if multipliers is None:
+            multipliers = len(self.objfcts)*[1]
+        else:
+            for mult in multipliers:
+                assert(type(mult) in self._multiplier_types), (
+                    "Objective Functions can only be multiplied by a float, or"
+                    " a properties.Float, not a {}, {}".format(
+                        type(mult), mult
+                    )
+                )
+            assert len(multipliers) == len(self.objfcts), (
+                "Length of multipliers ({}) must be the same as the length of "
+                "objfcts ({})".format(len(multipliers), len(self.objfcts))
+            )
+        self._multipliers = multipliers
+
+        super(ComboObjectiveFunction, self).__init__(**kwargs)
+        self._nP = nP
+
+    @property
+    def multipliers(self):
```

Comment:
or perhaps this is solved with properties? 

we do want to be able to change the multipliers - treating them as directives, but I agree there needs to be further thought into how we do this in a safe way